Project.toml

```
@@ -1,6 +1,6 @@
 name = "StaticArrays"
 uuid = "90137ffa-7385-5640-81b9-e52037218182"
-version = "1.6.5"
+version = "1.6.6"
```

```suggestion
version = "1.7.0"
```
This PR is a new feature "supports custom samplers" rather than a bug fix, so I think it would be better to bump the minor version.